packages/orchestration/src/cosmos-api.ts

```diff
@@ -219,80 +282,83 @@ export interface StakingAccountActions {
    * Delegate an amount to a validator. The promise settles when the delegation is complete.
    * @param validator - the validator to delegate to
    * @param amount  - the amount to delegate
-   * @returns void
+   * @param [opts] - transaction submission options
+   * @returns unknown
    */
   delegate: (
     validator: CosmosValidatorAddress,
     amount: AmountArg,
-  ) => Promise<void>;
+    opts?: CosmosActionOptions,
+  ) => Promise<unknown>;
 
   /**
    * Redelegate from one delegator to another.
    * Settles when the redelegation is established, not 21 days later.
    * @param srcValidator - the current validator for the delegation.
    * @param dstValidator - the validator that will receive the delegation.
    * @param amount - how much to redelegate.
+   * @param [opts] - transaction submission options
    * @returns
    */
   redelegate: (
     srcValidator: CosmosValidatorAddress,
     dstValidator: CosmosValidatorAddress,
     amount: AmountArg,
-  ) => Promise<void>;
+    opts?: CosmosActionOptions,
+  ) => Promise<unknown>;
 
   /**
    * Undelegate multiple delegations (concurrently). To delegate independently, pass an array with one item.
    * Resolves when the undelegation is complete and the tokens are no longer bonded. Note it may take weeks.
    * The unbonding time is padded by 10 minutes to account for clock skew.
    * @param delegations - the delegation to undelegate
+   * @param [opts] - transaction submission options
    */
   undelegate: (
     delegations: {
       amount: AmountArg;
       delegator?: CosmosChainAddress;
       validator: CosmosValidatorAddress;
     }[],
-  ) => Promise<void>;
+    opts?: CosmosActionOptions,
+  ) => Promise<unknown>;
 
   /**
    * Withdraw rewards from all validators. The promise settles when the rewards are withdrawn.
+   * @param [opts] - transaction submission options
    * @returns The total amounts of rewards withdrawn
    */
-  withdrawRewards: () => Promise<DenomAmount[]>;
+  withdrawRewards: (opts?: CosmosActionOptions) => Promise<DenomAmount[]>;
 
   /**
    * Withdraw rewards from a specific validator. The promise settles when the rewards are withdrawn.
    * @param validator - the validator to withdraw rewards from
+   * @param [opts] - transaction submission options
    * @returns
    */
-  withdrawReward: (validator: CosmosValidatorAddress) => Promise<DenomAmount[]>;
+  withdrawReward: (
+    validator: CosmosValidatorAddress,
+    opts?: CosmosActionOptions,
+  ) => Promise<DenomAmount[]>;
 }
 
 /**
  * Low level methods from IcaAccount that we pass through to CosmosOrchestrationAccount
  */
 
-export interface IcaAccountMethods {
+export interface IcaAccountImplMethods {
```

Comment:
this name is confusing. how is can it be both an `Impl` and an `interface`?

Reply:
I changed this to `IcaAccountBaseMethods`.